ql/src/java/org/apache/hadoop/hive/ql/parse/AlterTableExecuteSpec.java

```diff
@@ -165,19 +165,19 @@ public String toString() {
    * </ul>
    */
   public static class SetCurrentSnapshotSpec {
-    private final long snapshotId;
+    private final String snapshotIdOrRefName;
 
-    public SetCurrentSnapshotSpec(Long snapshotId) {
-      this.snapshotId = snapshotId;
+    public SetCurrentSnapshotSpec(String snapshotIdOrRefName) {
+      this.snapshotIdOrRefName = snapshotIdOrRefName;
     }
 
-    public Long getSnapshotId() {
-      return snapshotId;
+    public String getSnapshotIdOrRefName() {
+      return snapshotIdOrRefName;
     }
 
     @Override
     public String toString() {
-      return MoreObjects.toStringHelper(this).add("snapshotId", snapshotId).toString();
+      return MoreObjects.toStringHelper(this).add("snapshotId", snapshotIdOrRefName).toString();
```

Comment:
it can be either id or ref, can we change ``snapshotId`` to something common?

Reply:
`snapshotIdOrRefName` is ok?